### DISCUSSION TITLE
Is it possible to run VR on all devices with Rust? Are we heading towards the era of wearable VR PCs?

### DISCUSSION THREAD

**Discussion by PlumpMath** [bevyengine/bevy#11056] [Q&A]
Hello~*

Is it possible to make VR applications compatible with all devices using Rust? Are we approaching the era of wearable VR PCs? Rust is a great programming language. It seems to perform exceptionally well even at the lowest levels of hardware, but is there robust support for it in the VR development space? From my research, it doesn't appear to be the case.

https://www.reddit.com/r/rust_gamedev/comments/r68blf/is_it_possible_to_develop_an_oculus_quest_game_in/

It seems like the United States and Khronos standards are once again angling for a cutting-edge programming language. Our interactive cutting-edge developers need to be liberated on the web. This is particularly relevant now, with the recent release of Apple's Vision Pro.

**Comment by mockersf** [comment]
there is an effort for support of XR in Bevy, it's already working and you can check it out there: https://github.com/awtterpip/bevy_oxr

**Comment by PlumpMath** [comment]
I hope that the project survives well.
And I look forward to the day when smartphones and laptop PCs disappear with the emergence of meta and all glasses wearable computers.

I also hope that the web becomes simpler. Anyway, we need to make the world a healthier place.

**Comment by PlumpMath** [comment]
Meta should support as many program languages(& game engine) as vr owners use

Differences from smartphone OS

It doesn't seem like horion os is opened as the main one yet, but it seems like it's opened mainly as unity unreal. If you open more languages ​​later, the world economy will also improve.
